Lib/pathlib.py

```diff
@@ -1134,15 +1060,13 @@ def resolve(self, strict=False):
         normalizing it (for example turning slashes into backslashes under
         Windows).
         """
-        s = self._flavour.resolve(self, strict=strict)
-        if s is None:
-            # No symlink resolution => for consistency, raise an error if
-            # the path doesn't exist or is forbidden
-            self.stat()
-            s = str(self.absolute())
-        # Now we have no symlinks in the path, it's safe to normalize it.
-        normed = self._flavour.pathmod.normpath(s)
-        return self._from_parts((normed,))
+        try:
+            p = self._accessor.realpath(self, strict=strict)
+        except OSError as ex:
+            if ex.errno == ELOOP:
```

Comment:
The behavior in Windows could be changed to raise `RuntimeError` for `ERROR_CANT_RESOLVE_FILENAME` (1921), which would agree with the documentation. For example: `ex.errno == ELOOP or getattr(ex, 'winerror', 0) == 1921`.

It would be simpler if `winerror_to_errno()` in "PC/errmap.h" mapped `ERROR_CANT_RESOLVE_FILENAME` to `ELOOP`. But I'm not certain that it's always used equivalently to `ELOOP`. The only NT status code that maps to this Windows error code is `STATUS_REPARSE_POINT_NOT_RESOLVED`, which is due to exhausting the maximum allowed number of reparse attempts when opening a path (currently 63). But a system component could directly use `ERROR_CANT_RESOLVE_FILENAME` in a broader sense, in which case mapping it to `ELOOP` would be misleading.